### ISSUE TITLE
check for required secret/configmap/psp before creating the csi driver resources should be there [Improvement]

### ISSUE THREAD

**Issue by vrushch** [IBM/ibm-spectrum-scale-csi#551]
## Describe the bug
When we apply cr file without creating required secret/configmap/psp , then also driver pods are getting created

## How to Reproduce?
Please list the steps to help development teams reproduce the behavior

1. apply cr file with correct details without creating required secrets

## Expected behavior
Operator should check for availability of required secret/configmap/psp , then only create driver pods

**Comment by deeghuge** [comment]
@nitishkumar4 @amdabhad Is this still valid ?

**Comment by amdabhad** [comment]
Yes, we still don't have checks for the existence of these resources.
